src/travel-details-screens/components/Trip.tsx

```diff
@@ -25,19 +28,37 @@ export type TripProps = {
 const Trip: React.FC<TripProps> = ({
   tripPattern,
   error,
+  onPressDetailsMap,
   onPressDeparture,
   onPressQuay,
 }) => {
   const styles = useStyle();
   const legs = tripPattern.legs.filter((leg, i) =>
     isSignificantFootLegWalkOrWaitTime(leg, tripPattern.legs[i + 1]),
   );
+  const shouldShowLive = isDateInRangeFromNow(
+    tripPattern.expectedStartTime,
+    12 * 60,
+  );
+
+  const ids = shouldShowLive
+    ? (tripPattern.legs
+        .map((leg) => leg.serviceJourney?.id)
+        .filter(Boolean) as string[])
+    : undefined;
+  const {vehiclePositions} = useGetServiceJourneyVehicles(ids);
+
   return (
     <View style={styles.container}>
       <TripMessages tripPattern={tripPattern} error={error} />
       <View style={styles.trip}>
         {tripPattern &&
           legs.map((leg, index) => {
+            const tripVehiclePosition = vehiclePositions?.find(
```

Comment:
It is the position for the vehicle in a single leg, so I think `legVehiclePosition` is more precise.